src/util/simplify_expr_int.cpp

```diff
@@ -1125,18 +1152,38 @@ simplify_exprt::simplify_extractbits(const extractbits_exprt &expr)
       if(!op_width.has_value() || *op_width <= 0)
         return unchanged(expr);
 
-      if(*start + 1 == offset && *end + *op_width == offset)
+      if(*start < offset && *end >= offset - *op_width)
```

Comment:
NitPick: Prefer the following for readability:
`if(*start < offset && offset <= *end + *op_width)`
Its closer to the math shorthand of `*start < offset <= *end + *op_width`

Reply:
Done.